### PR TITLE
First working implementation

### PR DESCRIPTION
**Pull request by dmytro**
This is first implementation of Nagios plugin for monitoring expiration of OpenVPN certs. Some of the certs will be expiring soon, I had a calendar event about expiration of mine, so I didn't miss it this time :smiley_cat: But don't really want to depend on calendar next time.

It's built as a gem, will need also to add Chef recipe to include in deployment process a bit later.

@zentahori @shouichi @clownba0t 
Please review

### CODE REVIEW

**Comment by dmytro** [comment]
@zentahori @shouichi @clownba0t 
please note: this is a public repo.

File: version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+0.0.1
```
Comment (shouichi) [inline version.txt]: Normally gem version is specified as a constant I think.

File: openvpn_cert_nagios.gemspec [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  s.summary = 'Check expiration date of the OpenVPN certs'
+  s.description =<<-EOF
+
+For every certificate in EasyRSA directory check expiration date and issue warning or critial warning alert.
+
+EOF
+
+  s.required_ruby_version = '>= 2.0.0'
+
+  s.files = ( %w{ bin/check_open_vpn_cert version.txt} +
```
Comment (shouichi) [inline openvpn_cert_nagios.gemspec]: What about using git? For example, `s.files = `git ls-files`.split($/)`.

File: lib/openvpn_cert_nagios/base.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env ruby
```
Comment (shouichi) [inline lib/openvpn_cert_nagios/base.rb]: Is this line needed?

Comment (dmytro) [inline lib/openvpn_cert_nagios/base.rb]: No it's not, copy-pasted. Removing.

File: openvpn_cert_nagios.gemspec (hunk 2 above)
Comment (dmytro) [inline openvpn_cert_nagios.gemspec]: Don't really like backtics in the code :)

File: version.txt (hunk 1 above)
Comment (dmytro) [inline version.txt]: I tried to look for conventions regarding this right now. 

Some guides are indeed using `::VERSION` and `lib/bla/version.rb`file, but there's no recommendation or guideline regarding this and in many cases version is simply specified in `.gemspec`. Like [here](http://guides.rubygems.org/make-your-own-gem/) or [here](http://guides.rubygems.org/patterns/#semantic-versioning) for example.

Reason I'm doing it this way, is that I like to see version file at the top of the source tree, not hidden somewhere deep in the `./lib` or assigned somewhere in the code where nobody can find it.

Comment (dmytro) [inline version.txt]: Moved to `version.rb` file, although I don't think i like it :disappointed:

**Comment by shouichi** [comment]
:+1:

File: lib/openvpn_cert_nagios/shell.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class CertCheck
+  ##
+  # Execute shell commands. Simple popen3 wrapper, that returns either
+  # STDOUT or exit status, and aborts on non zero status.
+  #
+  class Shell
+    def self.exe(cmd, ignore_status: false, only_status: false)
```
Comment (clownba0t) [inline lib/openvpn_cert_nagios/shell.rb]: Sure thing! Sorry for the misread.

The only other thing which might be slightly cleaner/easier to understand would be to use a single `return_data` style variable which could be set to something like `:status` or `:output`, with the default being `:error_or_output`.

**Comment by zentahori** [comment]
:+1: Nice gem package!

File: lib/openvpn_cert_nagios/shell.rb (hunk 4 above)
Comment (dmytro) [inline lib/openvpn_cert_nagios/shell.rb]: @clownba0t 
Hmmm... Tried several different cases with variables `return_data` (as in your advise) or something like `status: ignore|only` But I don't seem to like too much both of them... :( 

So, for now I'v simply deleted parts of the logic, that is not used here. Will see if I have better idea in the future. :)

**Pull request merged**